<a href="https://colab.research.google.com/github/PedroHudson/PUC_MVP3/blob/main/MVP_Engenharia_de_Dados.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**MVP - Engenharia de Dados**



**1. Objetivo:**

Esse projeto tem como objetivo construir um Bando de Dados na Nuvem. O escopo e desenvolvimento do projeto envolve a busca, coleta, modelagem, carga e análise de dados.

**2. Definição do Problema:**

Para a elaboração desse projeto será utilizado um dataset com dados de faturamento de um escritório de advocacia de médio porte, cuja sede localiza-se na cidade de São Paulo.

O escritório tem 33 anos de história, com forte atuação na área empresarial, reconhecidos pelo mercado jurídico como especialistas nas áreas de processo civil, direito aplicado aos FIDC’s (fundos de investimento em direitos creditórios), negócios empresariais, mercado imobiliário, especialmente loteamentos e negócios fiduciários, além de processo tributário e processo do trabalho.

Visando a confidencialidade e proteção dos dados, seguindo as normas e diretrizes da LGPD (Lei Geral de Proteção de Dados Pessoais), todos os dados confidenciais do dataset foram previamente tratados e transformados, para a utilização das informações de modo ético e seguro, garantindo a segurança dos dados e as boas práticas de governança.

Iremos analisar o faturamento da empresa nos últimos 4 anos, demonstraremos o faturamento por ano, por área e por rubrica, visualizaremos também os maiores clientes por faturamento e analisaremos como foi o desempenho do escritório durante e pós pandemia de COVID-19.







**3. Importação das Bibliotecas e Configurações.**

In [35]:
# Importação do Pandas
import pandas as pd

# Importação do Numpy
import numpy as np

# Importação do Pyplot
import matplotlib.pyplot as plt

# Importação de pacotes Matplotlib
from matplotlib import cm

# Importação do Seaborn
import seaborn as sns

# Configuração para não exibir Warnings
import warnings
warnings.filterwarnings('ignore')

**4. Importação do Dataset.**

In [36]:
# Importando dados da url para um dataframe

# url a importar
url_dados = 'https://github.com/PedroHudson/PUC_MVP3/blob/main/Faturamento.xlsx?raw=true'

# carga do dataset através da url
dataset = pd.read_excel(url_dados)


**5. Análise Exploratória - Primeiras Visualizações e Análises.**

In [37]:
# Visualização da quantidade de instâncias e atributos do dataset
dataset.shape

(10397, 5)

*Nessa primeira análise verificamos que o dataset tem 5 atributos e 10397 instâncias.*

In [38]:
# Visualização da quantidade de linhas por coluna
dataset.count()

Data       10397
Cliente    10397
Valor      10397
Rubrica    10397
Área       10397
dtype: int64

*Todos os atributos possuem a mesma quantidade de instâncias.*

In [39]:
# Visualização dos tipos de dados
dataset.dtypes

Data        object
Cliente     object
Valor      float64
Rubrica     object
Área        object
dtype: object

*A visualização dos tipos de dados nos permite verificar que os atributos "Data", "Cliente", "Rubrica" e "Área" são do tipo object, já o atributo "Valor" é do tipo float64.*

In [40]:
# Visualização das primeiras linhas do dataframe
dataset.head()

Data                             Cliente   Valor  \
0  02/01/2020  Cobra Brasil Fomento Mercantil S.A  969.76   
1  02/01/2020  Cobra Brasil Fomento Mercantil S.A  436.39   
2  02/01/2020                  Grupo Alves Cintra   71.23   
3  02/01/2020                  Grupo Alves Cintra   33.77   
4  03/01/2020    Escritório Advogados Associados   500.00   

               Rubrica                    Área  
0         Sucumbência   Recuperação de Crédito  
1  Honorários de Êxito  Recuperação de Crédito  
2         Sucumbência   Recuperação de Crédito  
3         Sucumbência        Contencioso Cível  
4         Sucumbência        Contencioso Cível

In [41]:
# Visualização das últimas linhas do dataframe
dataset.tail()

Data                                Cliente    Valor  \
10392  31/08/2023                    Minas Finanças Ltda  3700.00   
10393  31/08/2023                   Auto Mecânica Mister  1998.31   
10394  31/08/2023  Jose Norberto Fomento Mercantil Ltda.   101.48   
10395  31/08/2023                       Oerca Torres S/A    17.79   
10396  31/08/2023                       Oerca Torres S/A   127.64   

                   Rubrica                             Área  
10392         Sucumbência            Recuperação de Crédito  
10393           Fee Mensal                           Gestão  
10394  Honorários de Êxito  Recuperação Judicial e Falência  
10395  Honorários de Êxito  Recuperação Judicial e Falência  
10396  Honorários de Êxito  Recuperação Judicial e Falência